File: riak/ruby/read/riak_read.rb
#! /usr/bin/ruby
# -*- coding: utf-8 -*-
# ------------------------------------------------------------------
#	riak_read.rb
#
#					Sep/22/2015
#
# ------------------------------------------------------------------
require 'restclient'
require 'json'
#
load '/var/www/data_base/common/ruby_common/text_manipulate.rb'
# ------------------------------------------------------------------
puts	"*** 開始 ***\n"

url_base = 'http://host_ubuntu:8098/riak/shimane'
url_in = url_base + '?keys=true'

json_string = RestClient.get url_in
data_aa=JSON.parse(json_string)


count = data_aa['keys'].length
puts count

dict_aa={}
data_aa['keys'].each {|key|
	url_city = url_base + '/' + key
	json_city = RestClient.get url_city
	unit_aa=JSON.parse(json_city)
	if (unit_aa != nil)
		then
		dict_aa[key]=unit_aa	
	end
	}
#
dict_display_proc (dict_aa)
puts	"*** 終了 ***\n"
# ------------------------------------------------------------------

File: riak/rhino/read/riak_read.js
#! /usr/bin/java org.mozilla.javascript.tools.shell.Main
// ------------------------------------------------------------------
//
//	riak_read.js
//
//						Apr/08/2013
//
// ------------------------------------------------------------------
load ("/var/www/data_base/common/rhino_common/text_manipulate.js");
// ------------------------------------------------------------------
print	("*** 開始 ***");

var url_base="http://host_ubuntu1:8098/riak/shimane";

dict_aa = riak_to_dict_proc (url_base);

dict_display_proc (dict_aa);

print	("*** 終了 ***");

// ------------------------------------------------------------------
function riak_to_dict_proc (url_base)
{
	var url_in=url_base + '?keys=true';

	var json_str = readUrl(url_in);

	var data_aa = JSON.parse (json_str);

	var dict_aa = new Object ();

	for (var it in data_aa.keys)
		{
		var key = data_aa.keys[it];
		var url_city = url_base + '/' + key;
		var json_city = readUrl(url_city);
		var city = JSON.parse (json_city);
		dict_aa[key] = city;
		}

	return	dict_aa;
}


// ------------------------------------------------------------------
